internal/cli/cli.go

```diff
@@ -150,16 +152,19 @@ func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
 	setupScheme()
 
 	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
-		Scheme:                     scheme,
-		Host:                       "0.0.0.0",
-		Port:                       viper.GetInt("webhook-listen-port"),
-		CertDir:                    viper.GetString("webhook-certs-dir"),
-		MetricsBindAddress:         fmt.Sprintf("0.0.0.0:%d", viper.GetInt("metrics-listen-port")),
+		Scheme: scheme,
+		WebhookServer: webhook.NewServer(webhook.Options{
+			Host:    "0.0.0.0",
+			Port:    viper.GetInt("webhook-listen-port"),
+			CertDir: viper.GetString("webhook-certs-dir"),
+		}),
+		Metrics: server.Options{
+			BindAddress: fmt.Sprintf("0.0.0.0:%d", viper.GetInt("metrics-listen-port")),
+		},
 		HealthProbeBindAddress:     ":8080",
 		LeaderElection:             true,
 		LeaderElectionID:           "argoslower",
 		LeaderElectionResourceLock: "leases",
-		DryRunClient:               dryRun,
```

Comment:
Dry-run is important; since this can be set from the command line and change the behaviuor the K8s Client that the manager uses; preventing writes from occurring on all of its operations.

It looks like the new structure this all got embeded into `client.Options`
e.g.
```
import (
	ctrlclient "sigs.k8s.io/controller-runtimei/pkg/client"
)
....

		Client: ctrlclient.Options{
			DryRun: dryRun,
		},
```

Reply:
re-added dryrun